ArduPlane/mode_auto.cpp

```diff
@@ -17,6 +17,8 @@ bool ModeAuto::_enter()
     
     if (plane.quadplane.available() && plane.quadplane.enable == 2) {
         plane.auto_state.vtol_mode = true;
+        // always zero forward throttle demand on entry into VTOL modes
+        quadplane.q_fwd_throttle = 0.0f;
```

Comment:
We do have a quadplane method that is called on all any change of mode, we could put that rest there, that would remove the need for it in all the enter functions (but would not be exactly the same behavior)

https://github.com/ArduPilot/ardupilot/blob/7d264f1700d1d931ec401774be71bc170f158157/ArduPlane/quadplane.cpp#L4422-L4437

Reply:
OK, will move as much as possible for mode entry setting of that variable to QuadPlane::mode_enter